plugins/woocommerce-blocks/packages/components/checkbox-control/validated-checkbox-control.tsx

```diff
@@ -0,0 +1,233 @@
+		}, [ customValidation ] );
+
+		const { validationError, validationErrorId } = useSelect(
+			( select ) => {
+				const store = select( validationStore );
+				return {
+					validationError: store.getValidationError( errorIdString ),
+					validationErrorId:
+						store.getValidationErrorId( errorIdString ),
+				};
+			},
+			[ errorIdString ]
+		);
+
+		const validateInput = useCallback(
+			( errorsHidden = true ) => {
+				const inputObject = inputRef.current || null;
+
+				if ( inputObject === null ) {
+					return;
+				}
+
+				if (
+					inputObject.checkValidity() &&
+					customValidationRef.current( inputObject )
+				) {
+					clearValidationError( errorIdString );
+					return;
+				}
+
+				setValidationErrors( {
+					[ errorIdString ]: {
+						message: getValidityMessageForInput(
+							label,
+							inputObject,
+							customValidityMessage
+						),
+						hidden: errorsHidden,
+					},
+				} );
+			},
+			[
+				clearValidationError,
+				errorIdString,
+				setValidationErrors,
+				label,
+				customValidityMessage,
+			]
+		);
+
+		// Allows parent to trigger revalidation.
+		useImperativeHandle(
+			forwardedRef,
+			function () {
+				return {
+					focus() {
+						inputRef.current?.focus();
+					},
+					revalidate() {
+						validateInput( false );
+					},
+				};
+			},
+			[ validateInput ]
+		);
+
+		/**
+		 * Validation on mount.
+		 */
+		useEffect( () => {
+			if ( validateOnMount ) {
+				validateInput( true );
+			}
+		}, [ validateOnMount, validateInput ] );
+
+		// Remove validation errors when unmounted.
+		useEffect( () => {
+			return () => {
+				clearValidationError( errorIdString );
+			};
+		}, [ clearValidationError, errorIdString ] );
+
+		if ( passedErrorMessage !== '' && isObject( validationError ) ) {
+			validationError.message = passedErrorMessage;
+		}
+
+		const hasError = validationError?.message && ! validationError?.hidden;
+
+		return (
+			<CheckboxControl
+				className={ clsx( className, {
+					'has-error': hasError,
+				} ) }
+				aria-invalid={ hasError === true }
+				id={ textInputId }
+				aria-errormessage={
+					// we're using the internal `aria-errormessage` attribute, calculated from the data store.
+					// If a consumer wants to overwrite the attribute, they can pass a prop.
+					showError && hasError && validationErrorId
+						? validationErrorId
+						: undefined
+				}
+				ref={ inputRef }
+				onChange={ useCallback(
+					( newValue ) => {
+						validateInput( false );
+						// Push the changes up to the parent component.
+						onChange( newValue );
+					},
+					[ onChange, validateInput ]
+				) }
+				ariaDescribedBy={ ariaDescribedBy }
+				checked={ checked }
+				title="" // This prevents the same error being shown on hover.
+				label={ label }
+				feedback={
+					showError && hasError ? (
+						<ValidationInputError
+							errorMessage={ passedErrorMessage }
+							propertyName={ errorIdString }
+							elementId={ errorIdString }
```

Comment:
```suggestion
							propertyName={ errorIdString }
```

Passing `errorMessage` and/or `elementId` causes the error to always be shown, only `propertyName` should be passed here.